.changeset/ten-cougars-pretend.md

```diff
@@ -0,0 +1,69 @@
+---
+'@astrojs/netlify': major
+---
+
+# Netlify Adapter v4 simplifies static + SSR deployments
+
+This update is a complete overhaul of the Netlify adapter.
+It simplifies the user-facing config, and resolves a number of bugs along the way.
+
+Here's what changes:
+
+## Netlify Context is automatically available via Locals
+
+In v3, you could use `netlify-edge-middleware.ts` to inject data from the Netlify context into your Astro locals.
+In v4, this file is no longer needed because the Netlify context is automatically made available via `Astro.locals.netlify.context`.
+You can use this context to access information about the user (like geolocation or IP address), your Netlify site (like deploy ID) or the request (like its request ID or the CDN region it's served from).
+
+**Action Required:**
+Remove the `netlify-edge-middleware.ts` or `netlify-edge-middleware.js` file.
+In your codebase, change all usage of locals injected through that file to use `Astro.locals.netlify.context` instead.
+
+### Image CDN
+
+v4 of this adapter integrates your Astro site with Netlify [Image CDN](https://docs.netlify.com/image-cdn/overview/).
+This allows transforming images on-the-fly without impacting build times.
+It's implemented using an [Astro Image Service](https://docs.astro.build/en/reference/image-service-reference/), and enabled by default.
+
+## On-Demand Builders are no longer supported
+
+On-Demand Builders (ODB) allows SSR-Rendered pages to be cached using a Time to Live (TTL) strategy.
+ODB continues to be supported by Netlify, but has since been replaced by the much more powerful
```

Comment:
```suggestion
ODB continues to be supported by Netlify
```
this contradicts the heading. If you want people using the replacement instead, just say that

Reply:
Fair point! I've made some changes to this in https://github.com/withastro/adapters/pull/84/commits/b63cfcb76a28af009c90de5bebd3a7e274753d23 and 
 https://github.com/withastro/adapters/pull/84/commits/13a765983f4abcf09906e7d0ec477998ceef29a6 that should capture more accurately what's happening. How does that read to you?